content/tutorials/realtime-attendance-system-iot.textile

```diff
@@ -0,0 +1,262 @@
+
+<a href="/images/tutorials/iot-attendance/circuit-diagram.png" target="_blank">
+  <img src="/images/tutorials/iot-attendance/circuit-diagram.png" style="width: 100%" alt="Circuit diagram">
+</a>
+
+MFRC522 - Huzzah Pins
+* RST  - 4
+* SDA   - 5
+* MOSI - 13 
+* MISO - 12 
+* SCK  - 14 
+* 3V - 3.3V
+* GND - GND
+
+Caution: If you try to connect the power source of the RFID card to 5V instead of 3.3V, it might fry up.
+
+h2(#step3-mqtt-ably). Step 3 - Code to publish messages to Ably over MQTT
+
+In this step, we'll start writing the code to publish messages to Ably via MQTT when the RFID scanner detects an access card.
+
+Fire up your Arduino IDE and create a new Sketch file. To begin with, you'll need to add the correct board that you will use so that all the essential libraries are accesible (like WiFi etc). 
+
+From your Arduino's 'Tools' menu, go to 'Board' -> 'Boards Manager' and search for @ESP8266@ and install it.
+
+<a href="/images/tutorials/iot-attendance/boards-manager.png" target="_blank">
+  <img src="/images/tutorials/iot-attendance/boards-manager.png" style="width: 100%" alt="Boards Manager">
+</a>
+
+Next, you need to include the libraries needed for this app. You can do this by going into Arduino's 'Sketch' menu. 
+
+Sketch -> Include library -> Manage libraries. 
+
+Search for @MFRC522@ and install it. You need this library to enable the communication between the IoT board and the RFID sensor. This library comes with the @SPI@ library as well that you need to set up a serial communication protocol to communicate with any peripheral devices, in this case the RFID sensor. 
+
+The last thing that's needed is an MQTT library. You can use one of the libraries from Arduino's library manager, but for this tutorial we'll use the one by Joel Gaehwiler. You can "download ZIP library":https://www.arduinolibraries.info/libraries/mqtt directly and then include it the usual way Sketch -> Include library -> Add .ZIP library. All these libraries come with a few example sketches that you check out. We'll write our app from scratch.
+
+In your sketch file, start by including the libraries as follows:
+
+```[cplusplus]
+#include <SPI.h>
+#include <MFRC522.h>
+#include <ESP8266WiFi.h>
+#include <MQTT.h>
+```
+
+Next, define the RST and SS pins per the wiring in the previous step, as shown below:
+
+```[cplusplus]
+#define RST_PIN   4
+#define SS_PIN    5
+```
+
+Next, define the WiFi name and password so the board can connect to the internet using these credentials. Add the following:
+
+```[cplusplus]
+const char ssid[] = "<YOUR WIFI NAME>";
+const char pass[] = "<YOUR WIFI PASSWORD>";
+```
+
+Don't forget to replace the credentials in the above statement with your own.
+
+Next, instantiate the libraries so they can be used in the code. Add the following:
+
+```[cplusplus]
+MFRC522 mfrc522(SS_PIN, RST_PIN);
+WiFiClientSecure net;
+MQTTClient client;
+```
+
+Now, inside the default @setup()@ method of your sketch, add the following:
```

Comment:
is it worth explaining a little what setup() does? (not sure if this is outside of the scope of the tutuorial) maybe it could be a comment in the code?

Reply:
Sounds good, a line or two should be fine, done!